runtime/kubernetes/client/kubernetes.go

```diff
@@ -1,43 +1,149 @@
+// Package client provides an implementation of a restricted subset of kubernetes API client
 package client
 
+import (
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/micro/go-micro/util/log"
+)
+
+const (
+	// https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L134
+	dns1123LabelFmt string = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"
+)
+
+var (
+	// DefaultImage is default micro image
+	DefaultImage = "micro/micro"
+	// ServiceRegexp is used to validate service name
+	ServiceRegexp = regexp.MustCompile("^" + dns1123LabelFmt + "$")
+)
+
 // Kubernetes client
 type Kubernetes interface {
```

Comment:
This would probably be good as 

```
type Kubernetes interface {
    Deployment() Deployment
    Service() Service
}

type Deployment interface {
    Create() ...
    Delete() ...
    ...
}

type Service interface {
    Create() ...
    Delete() ...
    ...
}
```

Reply:
Need to come up with some renaming as there are already types called `Service` and `Deployment` in `types.go` -- these are the core K8s object structs

Reply:
ServiceAPI interface?

Reply:
Wait, I'm testing a different idea now. Will push shortly. Might be better 🤞